crates/bevy_time/src/fixed.rs

```diff
@@ -0,0 +1,345 @@
+use bevy_ecs::world::World;
+use bevy_reflect::Reflect;
+use bevy_utils::Duration;
+
+use crate::{time::Time, virt::Virtual, FixedUpdate};
+
+/// The fixed timestep game clock following virtual time.
+///
+/// A specialization of the [`Time`] structure. **For method documentation, see
+/// [`Time<Fixed>#impl-Time<Fixed>`].**
+///     
+/// It is automatically inserted as a resource by
+/// [`TimePlugin`](crate::TimePlugin) and updated based on
+/// [`Time<Virtual>`](Virtual). The fixed clock is automatically set as the
+/// generic [`Time`] resource during [`FixedUpdate`](bevy_app::FixedUpdate)
+/// schedule processing.
+///
+/// The fixed timestep clock advances in fixed-size increments, which is extremely useful for
+/// writing logic (like physics) that should have consistent behavior, regardless of framerate.
+///
+/// The default [`timestep()`](Time::timestep) is 64 hertz, or 15625
+/// microseconds. This value was chosen for being a power of two,
+/// which losslessly convert into [`f32`] and [`f64`], while only being
```

Comment:
> Is this lossless `f32` / `f64` conversion really necessary? Have we shown that it solves a real problem? 

No, but the product of another float and a power-of-two is also free of rounding errors, which is nice. People will often repeatedly multiply by `time.delta_seconds()` for integration, and having no drift with the default setting is just nice. 

`FixedUpdate` exists for accurate and stable math as much as it does to get a handle on performance.

> I think we can't / shouldn't do less than the framerate / 60hz by default, as without interpolation that will result in less than framerate-smooth physics.

The case you're using as an example is not smooth. If you set the tick rate to match the expected framerate, then most frames will run either 0 or 2 fixed updates (if your frame times are stable), because the duration of any frame will never quite exactly equal the fixed timestep. It's a degenerate case in a sense.

There is no default we can choose that can effectively substitute for actual smoothing.[^1]

> But doing this also means we're doing more work per frame than will likely be necessary for most games.

Not quite. 64 TPS @ 60 FPS would actually most likely result in _fewer_ frames having to run 2 fixed updates. The vast majority of frames (assuming stable frame times) would run 1.

> One commonality is that fixed update Hz are generally less than or equal to the frame time update Hz.

I think part of what you're asking is "Do other engines have specific reasons for their defaults that we may be discounting?" and I think the answer to that is "No."

The most compelling reasoning I can think of is that if an engine only samples inputs on a new frame, then if most monitors refresh at 60Hz, not updating more than the monitor refreshes and not ticking more than input is collected might feel like a natural conclusion.

But we don't sample inputs.
(It just so happens that we _alias_ input events to a frame, but we're working on improving that, see #9087). 

> Unity defaults to 50Hz.

I can't find it anymore, but I remember reading through a Unity thread where a Unity employee asked around internally and reported that nobody remembered why it's 50Hz. It's not a particularly "interesting" value anyway. Sometimes Unity users complain about it not being 60Hz [because they think it being 60Hz will make things look smoother][1]. (It won't!)

(If I find the thread, I will link it here for posterity.)

> Unreal seems to use "sub stepping", which breaks the normal update delta into X "substeps".

Unreal doesn't have an equivalent of `FixedUpdate`. Without sub-stepping, it's just variable delta time. With sub-stepping enabled, frames longer than the _max_ step size are subdivided into steps `<=` that step size. But the vast majority of them will advance by variable amounts smaller than that step size.

> Godot defaults to 60Hz.

I assume it's just because it's the most familiar choice. It's still the most common refresh rate, and you can point at plenty of games that feature that rate, whereas Counter Strike and Valorant are the only two I can name off the top that feature power-of-two tick rates.

> What is the cost of doing exactly 60Hz?

There is nothing actually special about 60Hz. It is a good default, because it's pretty responsive. But it's just a familiar number. I feel the general preference partly comes from the pretty pervasive misunderstanding about what results in smooth rendering.

64Hz has nicer properties (properties more relevant to why you'd use a fixed timestep) while being negligibly higher than 60Hz. That's why [I have been strongly advocating for it as the default][2].

[^1]: You can maybe [fuzz things to lock onto a preset framerate][3], but that strategy does not generalize.

[1]: https://forum.unity.com/threads/digital-foundry-unity-fixedtimestep-callout.1178485/
[2]: https://github.com/bevyengine/bevy/pull/8964#discussion_r1258949625
[3]: https://medium.com/@tglaiel/how-to-make-your-game-run-at-60fps-24c61210fe75

Reply:
Having said all that, it'd be fine if 60Hz was the default. Most users wouldn't think twice about it, which is also nice in a way.

I just don't think it should be treated as something beneficial for rendering. The fixed timestep is supposed to be decoupled from the framerate.

Reply:
@maniwani Feels stronger about this than I do, but I'll give my take as well.

> Is this lossless f32 / f64 conversion really necessary? Have we shown that it solves a real problem?

I don't think it is strictly necessary, or that we would have shown that it solves a real problem. It's more just a nice bonus for an arbitrary value that is >60 Hz. Anybody who needs lossless division can set it themselves.

> What is the cost of doing exactly 60Hz?

Doing "exactly" 60 Hz may have a pathological failure case where every other frame has two fixed updates and every other frame has zero. I also personally dislike cases where something works okay most of the time, but then suddenly sometimes there is a missed physics update frame or doubled update frame and the game developer doesn't fix this because it's not visible enough. I'd much rather have something that shows up every frame so they have to fix it.

Ultimately, I have no strong opinions on the default value - 50, 60, 62, 64, 120, 144, 240, 360, anything goes - so feel free to put anything you feel best there, and I don't think it is a significantly worse choice.

Reply:
I've seen this pathological 0/2 behavior in my games when using a 60hz fixed timestep: I think it's one of the worst defaults we could pick as a result. 64 Hz is fine, but mostly arbitrary beyond avoiding that problem.

Reply:
IMO, the default has to be 64Hz or 60Hz.

> I think we can't / shouldn't do less than the framerate / 60Hz by default without interpolation that will result in less than framerate-smooth physics.

I share cart's conclusion that less than 60Hz is no good, but I think so for a different reason.

I don't think a number less than 60Hz can be reasonably justified. Users just expect 60Hz or better from games, in terms of responsiveness (input delay), so that's the minimum.

Reply:
Ok I'm convinced, provided the doc here is updated to include the arguments here (namely the "pathological 0/2 behavior of 60" ).